docs/migration-from-5-to-6.md

```diff
@@ -0,0 +1,82 @@
+# Migration Guide from v5.x to v6.x
+
+The main motivation for this is adding rule-of-hooks for react. (https://www.npmjs.com/package/eslint-plugin-react-hooks)
+
+The change from v5 to v6 includes two main items:
+
+- Upgraded `eslint-config-airbnb` to `v18` which comes with major breaking rule changes.
+- Dropped eslint v4 support
+
+# Breaking rules that have been disabled
```

Comment:
Maybe we should put this last as this is not the most important information?

Reply:
Sure thing that sounds sensible 😄

Reply:
Done